app/validation/validator.py

```diff
@@ -559,18 +568,30 @@ def _get_numeric_range_values(self, answer, answer_ranges):
         }
 
     def validate_duplicates(self, json_to_validate):
-        special_keys = ['id']
+        """
+        question_id & answer_id should be globally unique with some exceptions:
+            - within a block, ids can be duplicated across variants, but must still be unique outside of the block.
+        """
+
         duplicate_errors = []
 
-        for special_key in special_keys:
-            unique_items = []
+        items_per_block = defaultdict(set)
 
-            for value in self._parse_values(json_to_validate, special_key):
-                if value in unique_items:
-                    duplicate_errors.append(
-                        self._error_message('Duplicate {} found. value {}'.format(special_key, value)))
-                else:
-                    unique_items.append(value)
+        for path, value in self._parse_values(json_to_validate, 'id'):
+            if 'blocks' in path:
+                # Generate a string path and add it to the set representing the ids in that path
+                path_list = path.split('/')
+
+                block_path = path_list[:path_list.index('blocks') + 2]
+
+                string_path = '/'.join(block_path)
+                # Using a set for items_per_block allows duplicate ids within a specific block.
+                items_per_block[string_path].add(value)
+
+        duplicates = find_duplicates(items_per_block.values())
```

Comment:
I think this now only checks block level ids and below i.e. it's not checking group ids for uniqueness.

Reply:
Fixed